src/main/kotlin/frontend/config/ConfigViewModel.kt

```diff
@@ -43,162 +37,66 @@ class ConfigViewModel(currentConfig: Config = Config.debug) : ViewModel() {
+            maxMutations = maxMutations!!,
+            mutationVariant = mutationVariant!!,
+            genomeLength = genomeLength!!,
+            births = births,
+            deaths = deaths,
+            population = population,
+            plantDensity = plantDensity,
+            dailyAverageEnergy = dailyAverageEnergy,
+            dailyAverageAge = dailyAverageAge,
+            gens = gens,
+            genomes = genomes,
+            csvExportEnabled = csvExportEnabled,
+            filename = filename!!,
+            seed = seed!!
           )
+        } catch (e: InvalidFieldException) {
+          errorMessage.update { e.message ?: "Unknown config error" }
+          null
+        } catch (ignored: Exception) {
+          null
         }
       }.stateIn(this)
 
-      isValid = combine(mapGroup, plantGroup, animalGroup, genomeGroup, statisticsGroup) { args ->
-        args.none { it == null }
-      }.stateIn(this)
-
-      simulationConfig = combine(
-        mapGroup,
-        plantGroup,
-        animalGroup,
-        genomeGroup,
-        statisticsGroup,
-      ) { mapGroup, plantGroup, animalGroup, genomeGroup, statisticsConfig ->
-        isValid.value.ifTrue {
-          safeFieldInit(configError) {
-            Config(
-              mapWidth = mapGroup!!.mapWidth.value,
-              mapHeight = mapGroup.mapHeight.value,
-              initialPlants = plantGroup!!.initialPlants.value,
-              nutritionScore = plantGroup.nutritionScore.value,
-              plantsPerDay = plantGroup.plantsPerDay.value,
-              plantGrowthVariant = plantGroup.plantGrowthVariant.value,
-              initialAnimals = animalGroup!!.initialAnimals.value,
-              initialAnimalEnergy = animalGroup.initialAnimalEnergy.value,
-              satietyEnergy = animalGroup.satietyEnergy.value,
-              reproductionEnergyRatio = genomeGroup!!.reproductionEnergyRatio.value,
-              minMutations = genomeGroup.minMutations.value,
-              maxMutations = genomeGroup.maxMutations.value,
-              mutationVariant = genomeGroup.mutationVariant.value,
-              genomeLength = genomeGroup.genomeLength.value,
-              births = statisticsConfig!!.births.value,
-              deaths = statisticsConfig.deaths.value,
-              population = statisticsConfig.population.value,
-              plantDensity = statisticsConfig.plantDensity.value,
-              dailyAverageEnergy = statisticsConfig.dailyAverageEnergy.value,
-              dailyAverageAge = statisticsConfig.dailyAverageAge.value,
-              gens = statisticsConfig.gens.value,
-              genomes = statisticsConfig.genomes.value,
-              csvExportEnabled = statisticsConfig.csvExportEnabled.value,
-              filename = statisticsConfig.filename.value,
-              seed = 0
-            )
-          }
-        }
-      }.stateIn(this)
+      isValid = combine(simulationConfig) { (config) -> config != null }.stateIn(this)
```

Comment:
useless. Just filter `simulationConfig` flow if not null

Reply:
unnecessary parameter name, parameter parenthesis

Reply:
redudant `combine`. You can map/filter over `Flow`

Reply:
Just use docs
https://kotlinlang.org/docs/flow.html#intermediate-flow-operators
or 
https://kt.academy/article/cc-understanding-flow